frontend/packages/noobaa-storage-plugin/src/components/buckets-card/buckets-card-item.tsx

```diff
@@ -1,75 +1,114 @@
 import * as React from 'react';
-import * as _ from 'lodash';
+import { Link } from 'react-router-dom';
+import { InProgressIcon } from '@patternfly/react-icons';
 import { RedExclamationCircleIcon } from '@console/shared';
-import { LoadingInline, pluralize, humanizeNumber } from '@console/internal/components/utils';
+import { humanizeNumber, pluralize } from '@console/internal/components/utils';
 
 const formatCount = (count: number) => {
   const hCount = humanizeNumber(count);
   return `${hCount.string} Object${count === 1 ? '' : 's'}`;
 };
 
-const BucketsRowStatus: React.FC<BucketsRowStatusProps> = React.memo(({ status, link, error }) => (
-  <div className="nb-buckets-card__row-status-item">
-    {error || _.isNil(status) ? (
-      <span className="co-dashboard-text--small nb-buckets-card__row-subtitle">Not available</span>
-    ) : Number(status) > 0 ? (
-      <>
-        <a href={link} style={{ textDecoration: 'none' }} target="_blank" rel="noopener noreferrer">
-          <RedExclamationCircleIcon className="co-dashboard-icon nb-bucket-card__status-icon" />
-          <span className="nb-buckets-card__row-status-item-text">{status}</span>
-        </a>
-      </>
-    ) : null}
-  </div>
-));
+// Displays count of erroneous buckets due to issues not externalized as phase
+const OtherFailure: React.FC<BucketFailureItemProps> = React.memo(({ link, status }) => {
+  return (
+    <div className="nb-buckets-card__buckets-failure-status-item">
+      <a href={link} style={{ textDecoration: 'none' }} target="_blank" rel="noopener noreferrer">
+        <RedExclamationCircleIcon />
+        <span className="nb-buckets-card__buckets-failure-status-count">{status}</span>
+      </a>
+    </div>
+  );
+});
 
-const BucketsRow: React.FC<BucketsRowProps> = React.memo(
-  ({ bucketsCount, title, objectsCount }) => {
-    const subtitle: string = _.isNil(objectsCount)
-      ? 'Not available'
-      : formatCount(Number(objectsCount));
-    return (
-      <div className="nb-buckets-card__row-title">
-        <div>{_.isNil(bucketsCount) ? title : pluralize(Number(bucketsCount), title)}</div>
-        <div className="co-dashboard-text--small nb-buckets-card__row-subtitle">{subtitle}</div>
-      </div>
-    );
+// Displays count of erroneous buckets due to failure in provision
+const ProvisioningFailure: React.FC<BucketFailureItemProps> = React.memo(({ link, status }) => {
+  return (
+    <div className="nb-buckets-card__buckets-failure-status-item">
+      <Link to={link} style={{ textDecoration: 'none' }}>
+        <InProgressIcon className="co-inventory-card__status-icon--progress" />
+        <span className="nb-buckets-card__buckets-failure-status-count">{status}</span>
+      </Link>
+    </div>
+  );
+});
+
+// Displays count of erroneous buckets
+const BucketFailureStatus: React.FC<{ count: number[]; link: string[] }> = React.memo(
```

Comment:
I know it's very short and it's not externally used but then again it would be better to declare a type.

Reply:
ack